functions/utils/recording_events.py

```diff
@@ -0,0 +1,245 @@
+import logging
+from datetime import datetime, timedelta
+import boto3
+from botocore.exceptions import ClientError
+from os import getenv as env
+from dotenv import load_dotenv
+from os.path import join, dirname
+from .common import TIMESTAMP_FORMAT
+from pytz import utc
+from itertools import chain
+
+logger = logging.getLogger()
+
+load_dotenv(join(dirname(__file__), "../.env"))
+
+RECORDING_EVENTS_TABLE = env("RECORDING_EVENTS_TABLE", None)
+
+
+RECORDING_EVENTS = [
+    "recording.started",
+    "recording.paused",
+    "recording.resumed",
+    "recording.stopped",
+]
+
+START_EVENTS = ["recording.started", "recording.resumed"]
+END_EVENTS = ["recording.paused", "recording.stopped"]
+
+# This was copied from status.py:
+# If RECORDING_EVENTS_TABLE is not set we assume the events table
+# is not being used. (For example tasks.py imports utils but doesn't
+# use the recording times table)
+recording_events_table = None
+if RECORDING_EVENTS_TABLE:
+    dynamodb = boto3.resource("dynamodb")
+    recording_events_table = dynamodb.Table(RECORDING_EVENTS_TABLE)
+
+
+def get_recording_events(zoom_uuid):
+    """
+    Gets recording events from the database by zoom uuid.
+    """
+    r = recording_events_table.get_item(Key={"zoom_uuid": zoom_uuid})
+    logger.debug(f"Recording events from db: ${r}")
+    return r["Item"] if "Item" in r else None
+
+
+def set_recording_events(
+    zoom_uuid,
+    zoom_event,
+    zoom_event_timestamp,  # in ms
+    meeting_id=None,
+):
+    """
+    Records a recording event in the database. Events are stored by zoom uuid in a list of
+    {
+        "event": "ZOOM_EVENT",  # recording.started, paused, resumed, stopped
+        "timestamp": "TIMESTAMP_IN_SECS",
+    }
+    """
+    # If not one of desired events or no timestamp, ignore
+    if zoom_event not in RECORDING_EVENTS or not zoom_event_timestamp:
+        return
+
+    # If this is an event for recording started/paused/resumed/stopped,
+    # record the event time in the table in a list.
+    formatted_local_time = datetime.strftime(
```

Comment:
what's "local" about this?

Reply:
Local time instead of utc, just for clarity.

Reply:
oh ok. you're purposely not converting to UTC. That's fine for the purposes of logging something, but the comments just above are misleading.

Reply:
I've changed the comment to make it clear.